### PR TITLE
Add new utils; clean up docs

### PR DESCRIPTION
**Pull request by marcesher**
Adds two new utils and cleans up docs and some (not all) example code

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -31,128 +42,94 @@ new CheckmarxSecurityJobBuilder(
...
-            """
-    ).build(this);
-
-    customFlowJob.with{
-      logRotator{
-          numToKeep(365)
-      }
-
-    }
+FlowJobBuilder(
```
Comment (Scotchester) [inline docs/examples.md:61]: This needs a `new` in front of it, doesn't it?

File: docs/examples.md [hunk 2]
```diff
@@ -31,128 +42,94 @@ new CheckmarxSecurityJobBuilder(
...
+        name: 'sample-flow-job',
+        description: 'A sample build flow job that builds a list of jobs in order',
+        jobs:['job1', 'job2']
+).build(this).with {
+  logRotator{
+      numToKeep(365)
```
Comment (Scotchester) [inline docs/examples.md:67]: Is there a reason to encourage overriding our default of `numToKeep(30)` here and in the example below?

File: docs/examples.md [hunk 3]
```diff
@@ -31,128 +42,94 @@ new CheckmarxSecurityJobBuilder(
...
 import jenkins.automation.builders.JsJobBuilder
 
-
 String basePath = 'JsJobSamples'
 List developers = ['irina.muchnik@cfpb.gov', 'daniel.davis@cfpb.gov']
 
 def repos = [
         [name: 'jenkins-automation', url: "https://github.com/cfpb/jenkins-automation@2.0"],
-        [name: 'collab', url: "https://github.com/cfpb/jenkins-automation", shallow: true]
+        [name: 'collab', url: "https://github.com/cfpb/collab", shallow: true]
```
Comment (Scotchester) [inline docs/examples.md:99]: `shallow: true` is the default now; this could be removed.

File: docs/examples.md [hunk 4]
```diff
@@ -31,128 +42,94 @@ new CheckmarxSecurityJobBuilder(
...
         [name: 'other', url: "https://github.com/cfpb/jenkins-automation", disable_submodule: true]
```
Comment (Scotchester) [inline docs/examples.md:100]: What does the `name` in a `repos` item do?

File: docs/examples.md (hunk 1 above)
Comment (marcesher) [inline docs/examples.md:61]: good catch

File: docs/examples.md (hunk 2 above)
Comment (marcesher) [inline docs/examples.md:67]: I think it's useful to at least show how... I don't think that's exactly encouraging it

File: docs/examples.md (hunk 3 above)
Comment (marcesher) [inline docs/examples.md:99]: will do

File: docs/examples.md (hunk 4 above)
Comment (marcesher) [inline docs/examples.md:100]: good question. Nothing, near as I can tell

File: docs/examples.md [hunk 5]
```diff
@@ -31,128 +42,94 @@ new CheckmarxSecurityJobBuilder(
...
+        [name: 'collab', url: "https://github.com/cfpb/collab", shallow: true]
+]
+new BaseJobBuilder(
+        name: "sample-job-with-multiscm",
+        description: "A sample with multiple source control repositories",
+).build(this).with {
     multiscm {
-        ScmUtils.project_repos(delegate, this.repos, use_versions)
-        }
+        ScmUtils.project_repos(delegate, repos, true)
```
Comment (Scotchester) [inline docs/examples.md:130]: Our typical pattern is to [pass `repos` into the constructor](https://[GHE]/CFPB/cfgov-jenkins-automation/blob/master/jobs/cfgov-deploy-apache-conf.groovy#L16), which elimintates the need for the `multiscm` block in the `with`.

Should we do that in this example?

File: example-jobs/SampleBaseJobs.groovy [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+def repos = [
+        [name: 'jenkins-automation', url: "https://github.com/cfpb/jenkins-automation@2.0"],
+        [name: 'collab', url: "https://github.com/cfpb/collab", shallow: true]
+]
+new BaseJobBuilder(
+        name: "sample-base-job-with-multiscm",
+        description: "A sample with multiple source control repositories",
+).build(this).with {
+    multiscm {
+        ScmUtils.project_repos(delegate, repos, true)
```
Comment (Scotchester) [inline example-jobs/SampleBaseJobs.groovy:37]: Same comment about passing in `repos`

File: docs/examples.md (hunk 5 above)
Comment (marcesher) [inline docs/examples.md:130]: BaseJob does not support repos (nor, I think, should it)

File: docs/examples.md [hunk 7]
```diff
@@ -196,15 +174,40 @@ job('example'){
...
-         
+    CommonUtils.addExtendedEmail(delegate, emails = 'foo@example.com, bar@example.com') 
 }
    
-   
 // override accepts emails as a list. Compatible with builders
 job('example'){
-  CommonUtils.addExtendedEmail(delegate, 
-  emails = ['someperson@email.com', 'someotherperson@email.com']) 
+    CommonUtils.addExtendedEmail(delegate, emails = ['foo@example.com', 'bar@example.com'])
```
Comment (Scotchester) [inline docs/examples.md:182]: Similar to the question about `repos`, we have the `emails` abstraction that can be added in to the constructor. Should we document that pattern instead?

File: docs/examples.md (hunk 5 above)
Comment (Scotchester) [inline docs/examples.md:130]: Ah, right you are.

File: docs/examples.md (hunk 7 above)
Comment (Scotchester) [inline docs/examples.md:182]: NM, per the fact that baseJobBuilder doesn't support `emails`, either.

Comment (Scotchester) [inline docs/examples.md:182]: Gah, spoke too soon. It DOES have `emails` support. So I reopen this line of inquiry.

**Pull request merged**